### PR TITLE
test: Test for warning when distance column and distance_cm specified

### LINKED ISSUE #610: write test for distance warning
During tests a warning is raised for specifying both experiment.screen.distance_cm and a distance column.

We are treating warnings as errors from now on and need to get rid of the ignore in pyproject.toml

- [x] write test for warning
- [x] remove ignore from pyproject.toml

Here is the ignore line in
https://github.com/aeye-lab/pymovements/blob/417a305566646de08306a8682a9331e1c5f9ccae/pyproject.toml#L134

### PR DESCRIPTION
**Pull request by jakobchwastek**
fix #610

### CODE REVIEW

**Comment by jakobchwastek** [comment]
This test is failing now because we treat the warning it is triggering as error. Any suggestions on how to handle this? I would like to keep the test

https://github.com/aeye-lab/pymovements/blame/a95c1da4a23dbbb1f23229d90ed1203a21ace4f6/tests/unit/gaze/gaze_transform_test.py#L489

**Comment by dkrako** [comment]
> The `pix2deg_distance_experiment_and_distance_column_defaults_to_column` test is failing now because we treat the warning it is triggering as error. Any suggestions on how to handle this? I would like to keep the test
> 
> https://github.com/aeye-lab/pymovements/blame/a95c1da4a23dbbb1f23229d90ed1203a21ace4f6/tests/unit/gaze/gaze_transform_test.py#L489

Move the test case to ta new function and add the expected warning to the parameters:

```
def test_gaze_transform_expected_frame_warning(
        gaze_init_kwargs, transform_method, transform_kwargs, expected_result, expected_warning,
):
    with pytest.warns(expected_warning) as excinfo:
      gaze = pm.GazeDataFrame(**gaze_init_kwargs)
      gaze.transform(transform_method, **transform_kwargs)

    assert_frame_equal(gaze.frame, expected.frame)
```

File: tests/unit/gaze/gaze_transform_test.py [hunk 1]
```diff
@@ -844,14 +829,14 @@ def test_gaze_dataframe_pix2deg_exceptions(init_kwargs, exception, expected_msg)
         ),
     ],
 )
-def test_gaze_dataframe_pix2deg_warnings(init_kwargs, warning, expected_msg):
+def test_gaze_dataframe_pix2deg_exceptions(init_kwargs, exception, expected_msg):
     gaze_df = pm.GazeDataFrame(**init_kwargs)
 
-    with pytest.warns(warning) as excinfo:
+    with pytest.raises(exception) as excinfo:
```
Comment (dkrako) [inline tests/unit/gaze/gaze_transform_test.py:835]: I'm confused. Why aren't we using `pytest.warns` anymore?

**Comment by jakobchwastek** [comment]
Should be done

**Pull request merged**